presto-main/src/test/java/com/facebook/presto/execution/buffer/TestPageCodecMarker.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.buffer;
+
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.execution.buffer.PageCodecMarker.COMPRESSED;
+import static com.facebook.presto.execution.buffer.PageCodecMarker.ENCRYPTED;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestPageCodecMarker
+{
+    @Test
+    public void testCompressionAndEncryptionMarkers()
+    {
+        byte compressed = COMPRESSED.set(PageCodecMarker.none());
+        byte encrypted = ENCRYPTED.set(PageCodecMarker.none());
+        byte compressedAndEncrypted = ENCRYPTED.set(COMPRESSED.set(PageCodecMarker.none()));
+
+        // Not set when no markers present
+        assertFalse(COMPRESSED.isSet(PageCodecMarker.none()));
+        assertFalse(ENCRYPTED.isSet(PageCodecMarker.none()));
+
+        // Setting Markers
+        assertTrue(COMPRESSED.isSet(compressed));
+        assertTrue(COMPRESSED.isSet(compressedAndEncrypted));
+        assertTrue(ENCRYPTED.isSet(encrypted));
+        assertTrue(ENCRYPTED.isSet(compressedAndEncrypted));
+
+        // Unsetting Markers
+        assertEquals(COMPRESSED.unset(compressed), PageCodecMarker.none());
+        assertEquals(ENCRYPTED.unset(encrypted), PageCodecMarker.none());
+        assertFalse(COMPRESSED.isSet(COMPRESSED.unset(compressedAndEncrypted)));
+        assertFalse(ENCRYPTED.isSet(ENCRYPTED.unset(compressedAndEncrypted)));
+
+        // Summary String
+        assertEquals(PageCodecMarker.toSummaryString(PageCodecMarker.none()), "NONE");
+        assertEquals(PageCodecMarker.toSummaryString(encrypted), "ENCRYPTED");
+        assertEquals(PageCodecMarker.toSummaryString(compressed), "COMPRESSED");
+        assertEquals(PageCodecMarker.toSummaryString(compressedAndEncrypted), "COMPRESSED, ENCRYPTED");
+    }
+
+    @Test
+    public void testPageMarkerSanity()
```

Comment:
- This test tests `isSet` method, hence, let's rename it to `testIsSet`.
- s/m/marker
- drop messages from asserts; these tests are trivial and messages introduce unnecessary visual clutter


```
    @Test
    public void testIsSet()
    {
        assertEquals((byte) 0, PageCodecMarker.none());

        for (PageCodecMarker marker : PageCodecMarker.values()) {
            assertFalse(marker.isSet(PageCodecMarker.none()));
            assertTrue(marker.isSet(marker.set(PageCodecMarker.none())));
            assertFalse(marker.isSet(marker.unset(marker.set(PageCodecMarker.none()))));

            for (PageCodecMarker other : PageCodecMarker.values()) {
                assertEquals(other == marker, other.isSet(marker.set(PageCodecMarker.none())));
            }
        }
    }

    @Test
    public void testSummaryString()
    {
        assertEquals(PageCodecMarker.toSummaryString(PageCodecMarker.none()), "NONE");
        for (PageCodecMarker marker : PageCodecMarker.values()) {
            assertEquals(PageCodecMarker.toSummaryString(marker.set(PageCodecMarker.none())), marker.name());
        }

        byte allMarkers = PageCodecMarker.none();
        Arrays.stream(PageCodecMarker.values()).forEach(marker -> marker.set(allMarkers));
        
        String allMarkersSummary = PageCodecMarker.toSummaryString(allMarkers);
        for (PageCodecMarker marker : PageCodecMarker.values()) {
            assertTrue(allMarkersSummary.contains(marker.name()));
        }
    }
```

Reply:
Will do, unfortunately the usage of `Arrays.stream(PageCodecMarker.values()).forEach(...)` doesn't work since allMarkers needs to be reassigned which is prohibited for lambda closures (closed-over values must be final / effectively final). I'll build it in the loop above but otherwise keep this code.